components/clarinet-format/src/formatter/mod.rs

```diff
@@ -121,12 +130,22 @@ impl<'a> Aggregator<'a> {
     pub fn generate(&self) -> String {
         self.cache.borrow_mut().clear();
         // this handles if we're formatting a section of code rather than the whole file
-        let indentation_level = self
-            .source
-            .chars()
-            .take_while(|c| c.is_whitespace())
-            .count();
-        self.format_source_exprs(self.pse, &self.source[..indentation_level])
+        let indentation_level = match self.source {
+            Some(source) => source.chars().take_while(|c| c.is_whitespace()).count(),
+            None => {
+                (self
+                    .pse
+                    .first()
+                    .map_or(0, |expr| expr.span().start_column.saturating_sub(1)))
+                    as usize
+            }
+        };
+        let previous_indentation = match self.source {
+            Some(source) => &source[..indentation_level],
+            None => &" ".repeat(indentation_level),
+        };
```

Comment:
This is what I came up with following your previous review @tippenein 
This is a bit involved; I'm not sure we would encounter a use case where we need code section formatting without a text source. But for completeness, we have it

Reply:
It's possible I misunderstood what "without a text source" means. Looks ok to me